src/store/eventing/staged.ts

```diff
@@ -39,12 +35,11 @@ export const handleSetStagedAction = (prevState: AppState, nextState: AppState):
     const prevStaged = getCreatorStaged(prevState);
     const nextStaged = getCreatorStaged(nextState);
     const status = getStatus(prevStaged, nextStaged);
-    const type = getType(prevStaged) || getType(nextStaged);
-
-    if (!status || !type) {
+    const mode = getAnnotationMode(nextState);
+    const type = getType(prevStaged) || getType(nextStaged) || (mode === Mode.NONE ? null : mode);
```

Comment:
Does moving `(mode === Mode.NONE ? null : mode)` to the above line make it clear?
Also, emitting `'none'` as type is fine I think, because we have handled `'none'` case in FSM.

Reply:
yeah, seems like passing `mode` directly is fine, FSM doesn't even care about type when it processes a `cancel`

Reply:
For the first point, I think it looks okay to me? Could you show me what it would look like if I moved it above

For the second point, I think it would be cleaner if I keep it as is. If we can save on an extra call that is being to emitting none, I think that would be better in terms of performance.

Reply:
It rarely happens when staged changes but mode is none. Even if it happens, emitting an event, receiving in preview and preview doing nothing about it doesn't sound harmful for performance. Do you notice any measurable performance issues? I think in this case, cleaner code is preferred.

Reply:
Yeah performance probably isn't impacted here, highly unlikely `none` mode would result in a `staged` item change. By always returning a `type` we can get rid of one more conditional branch of code to maintain

Reply:
Sounds good I can clean up the logic here.